### DISCUSSION TITLE
AquaSec metrics in Prometheus

### DISCUSSION THREAD

**Discussion by nsmidhun12** [prometheus/prometheus#14603] [Q&A]
### What did you do?

Our AquaSec is installed in ECS, and we previously scraped the host in Prometheus, which indicates that the target is up and we are able to get the basic metrics (logins, scan -inprogress, pending, done). Therefore, there are no issues there. However, we are encountering a problem only with the metrics and need your suggestions on this issue.
When we click on the target in Prometheus or use the command 'curl -H "Authorization: Bearer <bearer token>" http://<hostname>/metrics`, we receive the following error: `{"message":"You are not allowed to perform this action","code":403}`.
The reason this happens is because no bearer token is supplied. We can test this by opening Developer Tools (ctrl + shift + I) and going to the Networking tab, doing the same thing.

please suggest

### What did you see instead? Under which circumstances?

{"message":"You are not allowed to perform this action","code":403}

<img width="588" alt="Screenshot 2024-08-02 085818" src="https://github.com/user-attachments/assets/c17f120c-4223-4fea-ac16-8a6ee9f624ab">
<img width="596" alt="Screenshot 2024-08-02 085838" src="https://github.com/user-attachments/assets/25f8a723-4fe4-4943-b2df-b4d63d651d00">

**Comment by machine424** [comment]
Are you using the shown URL or the one in the `href`? As sometimes they're not the same: https://github.com/prometheus/prometheus/issues/13376
As Prometheus can successfully scrape the target, it's clear this is not an issue in Prometheus itself.
You just need to better simulate the request that Prometheus runs.

I'll convert this into a discussion.

**Comment by nsmidhun12** [comment]
Hi, Thanks for the response, I am using the URL "http://console.aquasec.abc.com/metrics"

Also i have scraped the below in prometheus for aquasec,

  # AquaSec scrape configuration
  - job_name: "Aqua"
    scrape_interval: 5s
    scrape_timeout: 5s
    bearer_token: "xxxxxxxxxxxxxxxxxxxxxxxxx"
    scheme: http
    static_configs:
      - targets: ["console.aquasec.abc.com:80"]
